src/lib/accessibility/FocusZone/FocusTrapZone.tsx

```diff
@@ -0,0 +1,296 @@
+      this._root.current,
+      this._root.current.firstChild as HTMLElement,
+      true,
+    )
+    const _lastTabbableChild = getLastTabbable(
+      this._root.current,
+      this._root.current.lastChild as HTMLElement,
+      true,
+    )
+
+    if (ev.shiftKey && _firstTabbableChild === ev.target) {
+      focusAsync(_lastTabbableChild)
+      ev.preventDefault()
+      ev.stopPropagation()
+    } else if (!ev.shiftKey && _lastTabbableChild === ev.target) {
+      focusAsync(_firstTabbableChild)
+      ev.preventDefault()
+      ev.stopPropagation()
+    }
+  }
+
+  private _forceFocusInTrap = (ev: Event, triggeredElement: HTMLElement) => {
+    if (
+      FocusTrapZone._focusStack.length &&
+      this === FocusTrapZone._focusStack[FocusTrapZone._focusStack.length - 1]
+    ) {
+      if (!this._root.current.contains(triggeredElement)) {
+        this._findElementAndFocusAsync()
+        ev.preventDefault()
+        ev.stopPropagation()
+      }
+    }
+  }
+
+  private _handleOutsideFocus = (ev: FocusEvent): void => {
+    const focusedElement = document.activeElement as HTMLElement
+    focusedElement && this._forceFocusInTrap(ev, focusedElement)
+  }
+
+  private _handleOutsideClick = (ev: MouseEvent): void => {
+    const clickedElement = ev.target as HTMLElement
+    clickedElement && this._forceFocusInTrap(ev, clickedElement)
+  }
+
+  private _subscribeToEvents = () => {
+    const { forceFocusInsideTrap, isClickableOutsideFocusTrap } = this.props
+    if (forceFocusInsideTrap) {
+      eventStack.sub('focus', this._handleOutsideFocus, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+
+    if (!isClickableOutsideFocusTrap) {
+      eventStack.sub('click', this._handleOutsideClick, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+  }
+
+  private _unsubscribeFromEvents = () => {
+    const { forceFocusInsideTrap, isClickableOutsideFocusTrap } = this.props
+    if (forceFocusInsideTrap) {
+      eventStack.unsub('focus', this._handleOutsideFocus, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+
+    if (!isClickableOutsideFocusTrap) {
+      eventStack.unsub('click', this._handleOutsideClick, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+  }
+
+  private _getPreviouslyFocusedElementOutsideTrapZone = () => {
+    const { elementToFocusOnDismiss } = this.props
+    let previouslyFocusedElement = this._previouslyFocusedElementOutsideTrapZone
+
+    if (elementToFocusOnDismiss && previouslyFocusedElement !== elementToFocusOnDismiss) {
+      previouslyFocusedElement = elementToFocusOnDismiss
+    } else if (!previouslyFocusedElement) {
+      previouslyFocusedElement = document.activeElement as HTMLElement
+    }
+
+    return previouslyFocusedElement
+  }
+
+  private _hideContentFromAccessibilityTree = () => {
+    const bodyChildren = (document.body && document.body.children) || []
```

Comment:
lets explicitly put in comments somewhere (probably, in comments to `FocusTrapZone` class) the fact that this is a guy who is able to work in browser environment only